### ISSUE TITLE
[feature] pkgconfig generator: do not generate pc files for components if not explicitly registered

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan#7560]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

Generating pkg config files for each component might conflict pkg config files of others recipes, since:

- components names may be very generic. I think that most people expect that components can't conflict with others recipes. What would happen if pkg config name of a recipe is the same than the name of one component of another recipe, and a downtream recipe depends on both libs, use `pkg_config` generator but only care of first pkg config file?
- there might be a lot of components in a recipe (and this lib might not have official pkg config files, so it will only result in build folder cluttering with tones of useless .pc files).

Therefore, I suggest to generate "components pkg config" files only if it has been explicitly registered in upstream recipes with `self.cpp_info.components["component_name"].names["pkg_config"] = "pkgconfig_component_name"`

**Comment by memsharded** [comment]
Yes, this seems reasonable, I was not very happy about the high chance of collision, so probably this is something that recipes should somehow define (not sure if for every component, or some kind of global definition meaning "generate one .pc file per component or one that aggregates everything")

**Comment by danimtb** [comment]
I am closing this issue as outdated, as Conan 2 has evolved a lot regarding features and especially with the PkgConfig support.

There is a new PkgConfigDeps generator that handles the .pc generated files (even for components), so the name can be controlled: https://docs.conan.io/2/reference/tools/gnu/pkgconfigdeps.html#conan.tools.gnu.PkgConfigDeps

Also, this can be controlled by consumers as well by setting:

```
def generate(self):
    pc = PkgConfigDeps(self)
    # This will generate a my_new_component_name.pc for the component_name component of the dep_name dependency
    pc.set_property("dep_name::component_name", "pkg_config_name", "my_new_component_name")
    pc.generate()
```

See https://docs.conan.io/2/reference/tools/gnu/pkgconfigdeps.html#conan.tools.gnu.PkgConfigDeps.set_property fo reference.

If you have more questions regarding package config or any feature request, please feel free to open a new issue and we will be happy to help. Thank you!
